src/js/utils/url.js

```diff
@@ -131,16 +132,13 @@ export const getAbsoluteURL = function(url) {
  *           extension could be found.
  */
 export const getFileExtension = function(path) {
-  if (typeof path === 'string') {
-    const splitPathRe = /^(\/?)([\s\S]*?)((?:\.{1,2}|[^\/]+?)(\.([^\.\/\?]+)))(?:[\/]*|[\?].*)$/i;
-    const pathParts = splitPathRe.exec(path);
-
-    if (pathParts) {
-      return pathParts.pop().toLowerCase();
-    }
+  if (!path || typeof path !== 'string') {
+    return '';
   }
 
-  return '';
+  return extname(parseUrl(path).pathname)
```

Comment:
I am unsure if parseUrl is necessary here, but I think that this mostly deals with urls being in the url utils.